recipe/build.sh

```diff
@@ -2,14 +2,20 @@
 if [ "$target_platform" = "osx-arm64" ]; then
   export CMAKE_OSX_ARCHITECTURES="arm64"
 fi
-
+# fix scikit-build not finding the cross-compile python.
+if [ "$target_platform" != "$build_platform" ]; then
+  pyrootdir=$($PYTHON -c 'import sysconfig; print(sysconfig.get_path("stdlib"))')
+  crosspython="-DPython_ROOT_DIR=$pyrootdir -DPython_FIND_VIRTUALENV=ONLY"
```

Comment:
My point earlier is that scikitbuild uses the old way of finding python in cmake
https://github.com/scikit-build/scikit-build/blob/9787cc2582311f731b3d0411b1175bd852db2ad6/skbuild/resources/cmake/FindPythonExtensions.cmake#L248

https://cmake.org/cmake/help/latest/module/FindPythonLibs.html

The hints you are using, are only applicable to the new way of using find_python
https://cmake.org/cmake/help/latest/module/FindPython.html#hints

which scikit-build is hesitant to do https://github.com/scikit-build/scikit-build/pull/1019